.github/workflows/main.yml

```diff
@@ -0,0 +1,73 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
```

Comment:
I would prefer
```
  workflow_run:
      workflows: ["CI Tests"]
      branches: [master]
      types: 
        - completed
 ```
Then add this condition to the task
```
    if: ${{ github.event.workflow_run.conclusion == 'success' }}
````
You'll still be able to do manual deploys but the automated ones will only run if CI passes.